client/Client.ts

```diff
@@ -0,0 +1,56 @@
+import { app, BrowserWindow, ipcMain } from "electron";
+import IpcChannel from "./channels/IpcChannel.interface";
+import ExampleChannel from "./channels/ExampleChannel";
+
+/**Responsible for managing the Electron app (main process), the various windows (renderer processes), and IPC channels.*/
+export default class Client {
+  mainWindow: BrowserWindow | null; // first renderer process
+
+  constructor() {
+    app.on("ready", this.createMainWindow);
+    app.on("window-all-closed", this.onWindowAllClosed);
+    app.allowRendererProcessReuse = true;
+
+    this.registerIpcChannels();
+  }
+
+  /**Registers all the IPC channels for handling requests from the renderer process.*/
+  registerIpcChannels() {
+    const ipcChannels: IpcChannel[] = [new ExampleChannel()];
+
+    ipcChannels.forEach((channel) =>
+      ipcMain.on(channel.name, (
+        event,
+        argument?: string // TODO - generalize arg type
+      ) => channel.handle(event, argument))
+    );
+  }
+
+  private createMainWindow() {
+    this.mainWindow = new BrowserWindow({
+      // transparent: true,
+      // frame: false,

```

Comment:
Don't use these at all. In general, they create more problems than they solve. Very special use case.